### PR TITLE
177364340 bounty retrybutton

### PR DESCRIPTION
**Pull request by androidmaven**

### CODE REVIEW

**Comment by davkutalek** [comment]
This doesn't match the designed flow properly: After the USSD session has been recorded and finished the user should automatically be directed to the transaction details page, not stay on the bounty list. Submitted should not be a variable on the transaction model, it should just be information passed in the intent to the transaction details page. The transaction details page should always show the retry button if it is in the pending state.

**Review by davkutalek — CHANGES_REQUESTED** [review]
This doesn't match the designed flow properly: After the USSD session has been recorded and finished the user should automatically be directed to the transaction details page, not stay on the bounty list. Submitted should not be a variable on the transaction model, it should just be information passed in the intent to the transaction details page. The transaction details page should always show the retry button if it is in the pending state.

**Comment by androidmaven** [comment]
Thank you @davkutalek for the correction. The latest push commits addresses your comments.

**Review by davkutalek — APPROVED** [review]
LGTM, 2 minor comments.

File: app/src/main/res/values/logging_strings.xml [hunk 1]
```diff
@@ -59,6 +59,7 @@
 	<string name="clicked_copylink_request" translatable="false">Clicked copy link</string>
 	<string name="clicked_bounty_email_continue_btn" translatable="false">Clicked bounty email continue button</string>
 	<string name="clicked_run_bounty_session">Clicked run bounty USSD session button for account: $%1$s - $%2$s </string>
+	<string name="clicked_retry_bounty_session">Clicked retry bounty from transaction detail page: $%1$s </string>
```
[the remark this answers is not in the document]
Comment (androidmaven) [reply app/src/main/res/values/logging_strings.xml]: Got it, thanks!

File: app/src/main/java/com/hover/stax/bounties/BountyActivity.java [hunk 2]
```diff
@@ -32,26 +32,33 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
...
 	}
 
-	void makeCall(HoverAction a) {
+	public void makeCall(HoverAction a) {
 		Amplitude.getInstance().logEvent(getString(R.string.clicked_run_bounty_session, a.from_institution_name, a.root_code));
-		Intent i = new HoverParameters.Builder(this).request(a.public_id).setEnvironment(HoverParameters.MANUAL_ENV).buildIntent();
+		call(a.public_id);
+	}
+
+	public void retryCall(String actionId) {
```
[the remark this answers is not in the document]
Comment (androidmaven) [reply app/src/main/java/com/hover/stax/bounties/BountyActivity.java:45]: The flow wasn't added in XD, but will ask Justin

**Pull request merged**